pkg/controller/agent/pod.go

```diff
@@ -357,13 +357,50 @@ func applyRelatedEsAssoc(agent agentv1alpha1.Agent, esAssociation commonv1.Assoc
 		certificatesDir(esAssociation),
 	))
 
-	// Beats managed by the Elastic Agent don't trust the Elasticsearch CA that Elastic Agent itself is configured
-	// to trust. There is currently no way to configure those Beats to trust a particular CA. The intended way to handle
-	// it is to allow Fleet to provide Beat output settings, but due to https://github.com/elastic/kibana/issues/102794
-	// this is not supported outside of UI. To workaround this limitation the Agent is going to update Pod-wide CA store
-	// before starting Elastic Agent.
-	cmd := trustCAScript(path.Join(certificatesDir(esAssociation), CAFileName))
-	return builder.WithCommand([]string{"/usr/bin/env", "bash", "-c", cmd}), nil
+	// If agent is set to run as root, then we will add the Elasticsearch CA to the
+	// pod's trusted CA store as both FLEET_CA and ELASTICSEARCH_CA environment variables
+	// are not respected by Agent in fleet mode. If we're not running as root, then we'll document the procedure
+	// to add the Elasticsearch CA to the Kibana's xpack output configuration pertaining to Fleet.
+	//
+	// For historic purposes (https://github.com/elastic/beats/pull/26529). Agent didn't respect
+	// FLEET_CA until 7.14.0, which is the lowest valid version we support of Agent + Fleet.
+	if runningAsRoot(agent) {
+		cmd := trustCAScript(path.Join(certificatesDir(esAssociation), CAFileName))
+		return builder.WithCommand([]string{"/usr/bin/env", "bash", "-c", cmd}), nil
```

Comment:
If `FLEET_CA` is respected since `7.14.0`, why would we run `trustCAScript` based on `runningAsRoot` and not just version?

Reply:
Since this block is only called when running as fleet and we only support Agent + Fleet in 7.14.0+ checking the version would be duplicative as there's a version check prior to this that would error about "unsupported version".

https://github.com/elastic/cloud-on-k8s/blob/b56c06368cff9150ebc3067560b98119cbce5703/pkg/apis/agent/v1alpha1/validations.go#L50-L53
https://github.com/elastic/cloud-on-k8s/blob/b56c06368cff9150ebc3067560b98119cbce5703/pkg/controller/common/version/version.go#L34